source/selftest.py

```diff
@@ -0,0 +1,45 @@
+from utils import *
+import matplotlib.pyplot as plt
+
+config = load_config("config_test")
+
+import make
+import normalize
+import pipeline
+
+def selftest() :
+    """Test the implementation"""
+    try:
+        p = pipeline.pipeline(name="selftest")
+
+        # create the data artifacts used by the pipelines tasks
+        random = data("random")
+        random_copy = data("random_copy")
+        normal_max = data("normal_max")
+        normal_sum = data("normal_sum")
+
+        # create the pipeline tasks
+        p.add(make.random(outputs=[random]))
+        p.add(make.copy(inputs=[random],outputs=[random_copy]))
+        p.add(normalize.rows_max(inputs=[random],outputs=[normal_max]))
+        p.add(normalize.rows_sum(inputs=[random_copy],outputs=[normal_sum]))
+
+        # run the pipeline
+        p.run()
+
+        # plot desired data artifacts
+        normal_sum.plot("normal_sum.png",kind='area')
+
+        # transfer the data to the remote storage
+        p.save()
+
+    finally:
```

Comment:
Please don't ever silence an error - this is big red flag! Catch the exception - log it, handle it/ignore it, then move on. Doing things like these will break future implementation of auto crash reporting - giving us false positives.

Reply:
I don't understand. Exceptions are passed through (I know this because I got plenty before I got it working). Finally just makes sure the cleanup is done regardless of whether an exception is thrown.

Reply:
What I'm suggesting is that when this is running in the cloud, the exception doesn't get logged anywhere we can later find - all we'll know is that the pipeline stopped.